backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/steps/Sanitize.kt

```diff
@@ -0,0 +1,100 @@
+package app.ehrenamtskarte.backend.stores.importer.steps
+
+import app.ehrenamtskarte.backend.common.STATE
+import app.ehrenamtskarte.backend.stores.geocoding.isCloseTo
+import app.ehrenamtskarte.backend.stores.geocoding.queryFeatures
+import app.ehrenamtskarte.backend.stores.importer.PipelineStep
+import app.ehrenamtskarte.backend.stores.importer.types.AcceptingStore
+import kotlinx.coroutines.runBlocking
+import org.geojson.Feature
+import org.geojson.Point
+import org.slf4j.Logger
+
+const val DISTANCE_THRESHOLD_IN_KM = 1.0
+
+// Postal code lookup fails/does not really make sense for a "Postfach"
+const val STREET_EXCLUDE_PATTERN = "Postfach"
+
+class Sanitize(private val logger: Logger) : PipelineStep<List<AcceptingStore>, List<AcceptingStore>>() {
+
+    override fun execute(input: List<AcceptingStore>): List<AcceptingStore> = runBlocking {
+        val stateBbox = queryFeatures(listOf(Pair("state", STATE))).first().bbox
+
+        input.map { it.sanitize() }.filter {
+            if (!it.isInBoundingBox(stateBbox)) {
+                logger.info("'${it.name}, ${it.location}' was filtered out because it is outside of $STATE")
+                return@filter false
+            }
+            if (it.postalCode == null) {
+                // Probably because it is outside of the state but inside the bounding box of the state
+                logger.info("'${it.name}, ${it.location}' was filtered out because its postal code is null")
+                return@filter false
+            }
+            true
+        }
+    }
+
+    /**
+     * Sanitize the postal code and the coordinates of the [AcceptingStore] using forward geocoding.
+     * If the coordinates are not inside the bounding box of the postal code, one of those is wrong.
+     * Then query by the address and use the coordinates OR postal code of the first match to sanitize the store data.
+     */
+    private suspend fun AcceptingStore.sanitize(): AcceptingStore {
+        if (street?.contains(STREET_EXCLUDE_PATTERN) == true) return this
+
+        val storeInfo = listOfNotNull(name, location, street, houseNumber).joinToString()
+
+        val postalCodeBbox = if (postalCode != null) {
+            queryFeatures(listOf(Pair("postalcode", postalCode))).firstOrNull()?.bbox
+        } else null
+
+        // Postal code invalid
+        if (postalCodeBbox == null) {
+            val features = queryFeatures(this)
```

Comment:
I think it could make sense to call `queryFeatures` only once. Even though a cache is in place I think this class should not trust that the cache is working properly. Also disk io is also not that cheap.

Reply:
Actually this was only run once anyway since only one of both could be executed. Simplified the code though to remove duplicate.